utils/lister.cc

```diff
@@ -32,6 +32,8 @@ future<> lister::visit(directory_entry de) {
         }
 
         return _walker(_dir, std::move(de));
+    }).handle_exception_type([] (const std::runtime_error& e) {
```

Comment:
This looks too generic. runtime_error is a generic exception and you'll catch it if thrown anywhere, not just in the place you intend.

Maybe this should be in guarantee_type?

Reply:
I thought about it but did not want to change the semantic of  `guarantee_type`. Right now, if the type cannot be retrieved, it throws. Otherwise, type is always valid.

A different solution could be to squash `.then` and `.handle_exception_type` with `.then_wrapped`.